.github/workflows/ci.yml

```diff
@@ -76,33 +76,61 @@ jobs:
     outputs:
       os: ${{ steps.setup-os-matrix.outputs.os }}
 
-  build-and-test:
+  build:
+    runs-on: windows-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Setup the latest .NET 5 SDK
+        uses: actions/setup-dotnet@v1
+        with:
+          dotnet-version: 5.0.x
+
+      - name: Build
+        run: dotnet build --configuration=Release -p:TreatWarningsAsErrors=true Src
+
+  test:
     needs: setup-os-matrix
     strategy:
       matrix:
         os: ${{ fromJson(needs.setup-os-matrix.outputs.os) }}
         library: [ILGPU, ILGPU.Algorithms]
+        framework: [netcoreapp3.1, net5.0]
+        include:
+          # BUG in strategy matrix, workaround separate in two jobs
```

Comment:
is it a bug though? I think it's just the way it's supposed to work

Reply:
I would expect to work as we originally set it up, so that's why I put comment in this manner
it's not working only for the second option, for first (e.g ILGPU - first, ILGPU.Algoritham - second) it runs as expected but for a second it runs again with both values 
for example [test (windows-latest, ILGPU, ILGPU.Algorithms, net47)](https://github.com/ljubon/ILGPU/runs/3049145586?check_suite_focus=true#step:5:13) here we have both libraries, but we should have only second 🤸‍♂️

Reply:
What I mean is that I don't think it's supported to give an array of values for a matrix param in `include`. What is passed as the value for `library` in your example is `[ILGPU, ILGPU.Algorithms]` and that's it.

So essentially I think it's not a bug, but just a limitation of `include`.